test/e2e/proxy_test.go

```diff
@@ -1084,3 +1078,153 @@ func testProxyServiceLifeCycle(ipFamily *corev1.IPFamily, ingressIPs []string, d
+		wg.Add(1)
+		go func(index int, node string) {
+			defer wg.Done()
+			createAgnhostPod(t, data, fmt.Sprintf("agnhost-%d", index), node, false)
+		}(i, node)
+	}
+	wg.Wait()
+
+	testCases := []struct {
+		name                string
+		withSessionAffinity bool
+	}{
+		{
+			name:                "IPv4,withSessionAffinity",
+			withSessionAffinity: true,
+		},
+		{
+			name:                "IPv4,withoutSessionAffinity",
+			withSessionAffinity: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			skipIfNotIPv4Cluster(t)
+			ingressNodeIP := controlPlaneNodeIPv4()
+			ipProtocol := corev1.IPv4Protocol
+			lbIP := "1.1.2.1"
+			internalClientIP := internalClientIPs.ipv4.String()
+			externalClientIP := "1.1.1.1"
+			externalClientGateway := "1.1.1.254"
+			externalIPPrefix := 24
+
+			// Create another netns to fake an external network on the host network Pod.
+			externalClient := randName("external-client-")
+			cmd, externalNetns := getCommandInFakeExternalNetwork("sleep infinity", externalIPPrefix, externalClientIP, externalClientGateway)
+			if err := NewPodBuilder(externalClient, data.testNamespace, toolboxImage).OnNode(ingressNode).WithCommand([]string{"sh", "-c", cmd}).InHostNetwork().Privileged().Create(data); err != nil {
+				t.Fatalf("Failed to create external client: %v", err)
+			}
+			defer deletePodWrapper(t, data, data.testNamespace, externalClient)
+			if err := data.podWaitForRunning(defaultTimeout, externalClient, data.testNamespace); err != nil {
+				t.Fatalf("Error when waiting for Pod '%s' to be in the Running state", externalClient)
+			}
+
+			// Since the "external client" runs in a netns on ingress Node, we install a route on backend Node to route reply
+			// traffic to ingress Node.
+			addRouteToClientIPCmd := []string{"ip", "route", "replace", externalClientIP, "via", ingressNodeIP}
+			stdout, stderr, err := data.RunCommandFromAntreaPodOnNode(backendNode1, addRouteToClientIPCmd)
+			require.NoError(t, err, "Failed to add route to client IP on Node %s, stdout: %s, stderr: %s", backendNode1, stdout, stderr)
+			stdout, stderr, err = data.RunCommandFromAntreaPodOnNode(backendNode2, addRouteToClientIPCmd)
+			require.NoError(t, err, "Failed to add route to client IP on Node %s, stdout: %s, stderr: %s", backendNode2, stdout, stderr)
+			defer func() {
+				delRouteToClientIPCmd := []string{"ip", "route", "del", externalClientIP, "via", ingressNodeIP}
+				stdout, stderr, err := data.RunCommandFromAntreaPodOnNode(backendNode1, delRouteToClientIPCmd)
+				require.NoError(t, err, "Failed to delete route to client IP on Node %s, stdout: %s, stderr: %s", backendNode1, stdout, stderr)
+				stdout, stderr, err = data.RunCommandFromAntreaPodOnNode(backendNode2, delRouteToClientIPCmd)
+				require.NoError(t, err, "Failed to delete route to client IP on Node %s, stdout: %s, stderr: %s", backendNode2, stdout, stderr)
```

Comment:
I would use `assert` instead of `require` in deferred cleanup functions to avoid potentially skipping other cleanup tasks in case of an error